dask/dataframe/tests/test_groupby.py

```diff
@@ -1080,6 +1081,11 @@ def test_aggregate_dask():
         assert_max_deps(agg_dask1, 2)
         assert_max_deps(agg_dask2, 2)
 
+        # Make sure dict-based aggregation specs result in an
+        # explicit `getitem` layer to improve column projection
+        if isinstance(spec, dict):
+            assert hlg_layer(result1.dask, "getitem")
```

Comment:
I like that this `assert` is generic, but I do wonder if it's too generic and could still continue to pass if we somehow loose blockwise + getitem fusion in the future. Thoughts on how we might be able to more explicitly test that the column projection we're after is indeed happening?

Reply:
This is a good point. I did consider adding another getitem-fusion test, but it seemed to me like we already have decent coverage for the getitem optimization. I decided that the "important" change here is that the getitem layer is created, but I'm open to improving the coverage.